inferno-vc/src/Inferno/VersionControl/Operations.hs

```diff
@@ -340,87 +339,63 @@ fetchVCObjectHistory h = do
     head_h <- fetchCurrentHead h
     let head_fp = storePath </> "heads" </> show head_h
     preds <- readVCObjectHashTxt head_fp
-    pure $ head_h : preds
-  -- When we fold the history, we check if they exist in two places
-  -- 1. in 'vc_store' for available scripts
-  -- 2. then in 'vc_store/removed' for scripts that have been deleted
-  -- If a script has been deleted, we track its hash.
-  -- Since objects can never be modified, only deleted, we don't need to hold a lock here
-  let f acc hsh = do
-        existsInRoot <- liftIO $ doesFileExist $ storePath </> show hsh
-        existsInRemoved <- liftIO $ doesFileExist $ storePath </> "removed" </> show hsh
-
-        if existsInRoot
-          then do
-            obj <- fmap (const hsh) <$> fetchVCObject hsh
-            pure ((obj : fst acc), snd acc)
-          else do
-            if existsInRemoved
-              then do
-                obj <- fmap (const hsh) <$> fetchRemovedVCObject hsh
-                pure ((obj : fst acc), (hsh : snd acc))
-              else -- This script no longer exists even in 'removed' directory. The directory might get cleaned up by accident or something.
-              -- There are two choices we can make,
-              -- 1. Return a `VCMeta VCObjectHash` with dummy data
-              -- 2. Ignore this meta.
-              -- Approach no. 2 is taken here by just returning the accumulator.
-                pure acc
-  (metas, removeds) <- foldM f ([], []) history
-  -- The rest of this function handles the case when this script history was obtained by
-  -- cloning, and adds the original script to the returned history, if it still exists:
-  case removeds of
-    [] ->
-      -- if it is a clone, we would like to prepend source of the cloned script as part of the history.
-      -- it is fine to only do this once since we only show the last source of the clone
-      -- i.e. original -> cloneof orignal = cloned -> cloneof cloned = cloned'
-      -- when viewing cloned' history, it will only show up to cloned.
-      case metas of
-        all'@(x : _) ->
```

Comment:
This pattern match is looking at the second newest version, not the oldest version as intended.